tiledb/sm/cpp_api/ndrectangle.h

```diff
@@ -269,6 +269,36 @@ class NDRectangle {
   std::shared_ptr<tiledb_ndrectangle_t> ptr() const {
     return ndrect_;
   }
+  /**
+   * Get the data type of the range at idx
+   *
+   * @param dim_idx The dimension index.
+   * @return The datatype of the range.
+   */
+  tiledb_datatype_t range_dtype(unsigned dim_idx) {
```

Comment:
Should we actually make the C++ APIs public? Are there use cases for them outside of helping with type validation in the range getters and setters? I think those should be better served by an API to get the whole domain.

Reply:
@teo-tsirpanis in Slack/Shortcut our discussion was to not expose the entire domain, but rather, subaccessors such as `range_dtype` and `ndim` individually.